keyboards/otd356mini/keymaps/default/keymap.c

```diff
@@ -0,0 +1,193 @@
+/* Copyright 2019 Michael Doo
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// DIP switch bit alias
+#define BIT(x) (1UL<<x)
+
+extern keymap_config_t keymap_config;
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name
+enum otd_layers {
+  _QWERTY,
+  _MAC,
+  _DVORAK,
+  _COLEMAK,
+  _FUNCTION
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  /* Keymap _QWERTY: Base Layer (default)
+     * ,-----------------------------------------------------------.
+     * |Esc|  1|  2|  3|  4|  5|  6|  7|  8|  9|  0|  -|  =|    BSp|
+     * |-----------------------------------------------------------|
+     * |Tab  |  Q|  W|  E|  R|  T|  Y|  U|  I|  O|  P|  [|  ]|    \|
+     * |-----------------------------------------------------------|
+     * |Caps   |  A|  S|  D|  F|  G|  H|  J|  K|  L|  ;|  '|Enter  |
+     * |-----------------------------------------------------------|
+     * |Shift   |  Z|  X|  C|  V|  B|  N|  M|  ,|  .|  /|Shift     |
+     * |-----------------------------------------------------------'
+     * |Ctrl |Gui|Alt  |         Space         |Alt  |  Fn   |Ctrl |
+     * `-----------------------------------------------------------'
+     */
+  [_QWERTY] = LAYOUT(
+      KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,          KC_6,
+      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,          KC_Y,
+      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,          KC_H,
+               KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,          KC_B,
+      KC_LCTL, KC_LGUI, KC_LALT,
+      KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,        KC_BSPC,
+      KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,       KC_BSLS,
+      KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,        KC_NO,
+      KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,       KC_NO,
+      KC_SPC,                    KC_NO,   KC_RALT, MO(_FUNCTION), KC_RCTL),
```

Comment:
The purpose of the layout macro is to convert from the logical layout to the physical.
I don't understand why 7/8/9/0 are below CTL/GUI/ALT -- 7 should be after 6. All others should be updated accordingly.